lib/graphql/static_validation/rules/variable_usages_are_allowed.rb

```diff
@@ -7,32 +7,35 @@ class VariableUsagesAreAllowed
       def validate(context)
         # holds { name => ast_node } pairs
         declared_variables = {}
-
         context.visitor[GraphQL::Language::Nodes::OperationDefinition] << ->(node, parent) {
           declared_variables = node.variables.each_with_object({}) { |var, memo| memo[var.name] = var }
         }
 
         context.visitor[GraphQL::Language::Nodes::Argument] << ->(node, parent) {
-          return if !node.value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
-          arguments = nil
-
-          case parent
-          when GraphQL::Language::Nodes::Field
-            arguments = context.field_definition.arguments
-          when GraphQL::Language::Nodes::Directive
-            arguments = context.directive_definition.arguments
-          when GraphQL::Language::Nodes::InputObject
-            arg_type = context.argument_definition.type.unwrap
-            if arg_type.is_a?(GraphQL::InputObjectType)
-              arguments = arg_type.input_fields
+          node_values = Array.wrap(node.value)
+          return if !(node_values.all? { |value| value.is_a? GraphQL::Language::Nodes::VariableIdentifier })
```

Comment:
I tried to add a failing test similiar to your example, but it passess 🙈 

Could you help me write a red test?